lib/CustomFields/pages/EditCustomFieldsSettings/tests/DeleteCustomFieldsOptions-test.js

```diff
@@ -0,0 +1,260 @@
+          expect(editCustomFields.customFields(0).options(1).deleteButtonIsDisplayed).to.be.false;
+        });
+
+        describe('and the save button was clicked', () => {
+          beforeEach(async () => {
+            await editCustomFields.save();
+          });
+
+          it('should display the confirmation modal', () => {
+            expect(editCustomFields.deleteModal.modalIsPresent).to.be.true;
+          });
+
+          it('should display an option deletion warning in the modal', () => {
+            expect(editCustomFields.deleteModal.optionWarningMessages().length).to.equal(1);
+          });
+
+          describe('and the cancel button was clicked', () => {
+            beforeEach(async () => {
+              await editCustomFields.deleteModal.cancelButton.click();
+            });
+
+            it('should hide the confirmation modal', () => {
+              expect(editCustomFields.deleteModal.modalIsPresent).to.be.false;
+            });
+
+            it('should restore the option that was removed', () => {
+              expect(editCustomFields.customFields(0).options().length).to.equal(3);
+            });
+          });
+        });
+
+        describe('and the custom field was removed', () => {
+          beforeEach(async () => {
+            await editCustomFields.customFields(0).delete();
+          });
+
+          describe('and the save button was clicked', () => {
+            beforeEach(async () => {
+              await editCustomFields.save();
+            });
+
+            it('should not display an option deletion warning in the modal', () => {
+              expect(editCustomFields.deleteModal.optionWarningMessages().length).to.equal(0);
+            });
+
+            it('should display a custom field deletion warning in the modal', () => {
+              expect(editCustomFields.deleteModal.customFieldsWarningMessages().length).to.equal(1);
+            });
+
+            describe('and the cancel button was clicked', () => {
+              beforeEach(async () => {
+                await editCustomFields.deleteModal.cancelButton.click();
+              });
+
+              it('should hide the confirmation modal', () => {
+                expect(editCustomFields.deleteModal.modalIsPresent).to.be.false;
+              });
+
+              it('should restore the custom field that was removed', () => {
+                expect(editCustomFields.customFields().length).to.equal(1);
+              });
+
+              it('should restore the option that was removed', () => {
+                expect(editCustomFields.customFields(0).options().length).to.equal(3);
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe('when there is a custom field with used options', () => {
+    beforeEach(async function () {
+      this.server.get('/custom-fields', () => ({
+        customFields: [{
```

Comment:
I would recommend moving this object to a separate value. You use almost the same object earlier, so it would be better to minimize this point